src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs

```diff
@@ -447,11 +443,419 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                     }
 
                 case State.ResourceInside:
+                {
```

Comment:
Presumably this is just stuff moved from the other invoker. I don't need to fine-grain review it, right?